pkg/kubelet/pod/pod_manager_test.go

```diff
@@ -37,131 +40,179 @@ func newTestManager() (*basicManager, *podtest.FakeMirrorClient) {
+				Name:      "normal-pod-name",
+				Namespace: metav1.NamespaceDefault,
+			},
+		}
+	)
+
+	testCase := []struct {
+		name             string
+		podList          []*v1.Pod
+		needToRemovePod  *v1.Pod
+		expectPods       []*v1.Pod
+		expectMirrorPods []*v1.Pod
+	}{
+		{
+			name:             "Remove mirror pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  mirrorPod,
+			expectPods:       []*v1.Pod{normalPod, staticPod},
+			expectMirrorPods: []*v1.Pod{},
+		},
+		{
+			name:             "Remove static pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  staticPod,
+			expectPods:       []*v1.Pod{normalPod},
+			expectMirrorPods: []*v1.Pod{mirrorPod},
+		},
+		{
+			name:             "Remove normal pod",
+			podList:          []*v1.Pod{mirrorPod, staticPod, normalPod},
+			needToRemovePod:  normalPod,
+			expectPods:       []*v1.Pod{staticPod},
+			expectMirrorPods: []*v1.Pod{mirrorPod},
 		},
-		staticPod,
 	}
-	updates := append(expectedPods, mirrorPod)
-	podManager, _ := newTestManager()
-	podManager.SetPods(updates)
 
-	podManager.RemovePod(staticPod)
+	for _, test := range testCase {
+		t.Run(test.name, func(t *testing.T) {
+			podManager, _ := newTestManager()
+			podManager.SetPods(test.podList)
+			podManager.RemovePod(test.needToRemovePod)
+			actualPods1 := podManager.GetPods()
+			actualPods2, actualMirrorPods, _ := podManager.GetPodsAndMirrorPods()
 
-	actualPods := podManager.GetPods()
-	if len(actualPods) == len(expectedPods) {
-		t.Fatalf("Run RemovePod() error, expected %d pods, got %d pods; ", len(expectedPods)-1, len(actualPods))
-	}
+			sortOpt := cmpopts.SortSlices(func(a, b *v1.Pod) bool { return a.Name < b.Name })
 
-	_, _, orphanedMirrorPodNames := podManager.GetPodsAndMirrorPods()
-	expectedOrphanedMirrorPodNameNum := 1
-	if len(orphanedMirrorPodNames) != expectedOrphanedMirrorPodNameNum {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected %d orphaned mirror pods, got %d orphaned mirror pods; ", expectedOrphanedMirrorPodNameNum, len(orphanedMirrorPodNames))
-	}
+			// Check if the actual pods and mirror pods match the expected ones.
+			if diff := cmp.Diff(actualPods1, actualPods2, sortOpt); diff != "" {
+				t.Errorf("actualPods1 and actualPods2 are not equal: %s", diff)
+			}
+			if diff := cmp.Diff(actualPods1, test.expectPods, sortOpt); diff != "" {
+				t.Errorf("actualPods1 and expectPods are not equal: %s", diff)
 
-	expectedOrphanedMirrorPodName := mirrorPod.Name + "_" + mirrorPod.Namespace
-	if orphanedMirrorPodNames[0] != expectedOrphanedMirrorPodName {
-		t.Fatalf("Run getOrphanedMirrorPodNames() error, expected orphaned mirror pod name : %s, got orphaned mirror pod name %s; ", expectedOrphanedMirrorPodName, orphanedMirrorPodNames[0])
+			}
+			if diff := cmp.Diff(actualMirrorPods, test.expectMirrorPods, sortOpt); diff != "" {
+				t.Errorf("actualMirrorPods and expectMirrorPods are not equal: %s", diff)
+			}
```

Comment:
I'd suggest to use testify APIs, e.g.
```suggestion
                         assert.Empty(cmp.Diff(test.expectMirrorPods, actualMirrorPods, sortOpt)
```